src/Utility/Hash.php

```diff
@@ -161,7 +161,9 @@ public static function extract(array $data, $path)
             if ($conditions) {
                 $filter = [];
                 foreach ($next as $item) {
-                    if (is_array($item) && static::_matches($item, $conditions)) {
+                    if ((is_array($item) || $item instanceof ArrayAccess) &&
+                        static::_matches($item, $conditions)
```

Comment:
Thanks, I was mostly concerned with the initial parameter. Internal traversal is always a grab bag of data.